### ISSUE TITLE
Label Shape

### ISSUE THREAD

**Issue by jellyfishovo** [TaoShi1998/SSLCL#6]
Hi, @TaoShi1998 
Very interesting work! I’ve been trying to add this loss to my model recently, but I encountered the following issue: 

Based on the code, I am unable to get the correct shape for the label embedding. My true labels have the shape of `(bs * seq_length, 1)`, but using the `LabelEmbedding` class, the resulting embedding shape is `(bs * seq_length, feature_dim)` instead of `(num_classes, feature_dim)`. I reviewed the code but couldn’t understand this line:

```python
self.embedding_layer = nn.Embedding(num_classes, hidden_dim)
self.linear_layer = nn.Linear(hidden_dim, output_dim)
```

According to the logic of the code, the final shape should indeed be `(bs * seq_length, output_dim)` instead of `(num_classes, feature_dim)`. Is there an error in my input? How should I modify it so that the code works as expected?

**Comment by LuckyDaydreamer** [comment]
Hi @jellyfishovo, 

thank you for your interest in our work. In this code `self.embedding_layer = nn.Embedding(num_classes, hidden_dim)`, each class (assuming we are dealing with an N-class classification problem) is embedded into a dense representation with a dimension of `hidden_dim`. This is the same for every batch of labels, as the number of available labels is always the same for a given dataset. Therefore, `(bs * seq_length, output_dim)` should not be the shape of the embedded labels.

To modify this, simply replace `num_classes` with the number of classes relevant to your task (e.g., CIFAR-10 should be 10, ImageNet should be 1000) and choose an appropriate hidden dimension. But since I do not know the specific task you are working on, it might require additional adjustments.

Please feel free to ask if you have any questions.

Best,
Tao
